pkg/controller/endpointslice/endpointslice_controller.go

```diff
@@ -422,7 +437,11 @@ func (c *Controller) onEndpointSliceUpdate(prevObj, obj interface{}) {
 func (c *Controller) onEndpointSliceDelete(obj interface{}) {
 	endpointSlice := getEndpointSliceFromDeleteAction(obj)
 	if endpointSlice != nil && managedByController(endpointSlice) && c.endpointSliceTracker.Has(endpointSlice) {
-		c.queueServiceForEndpointSlice(endpointSlice)
+		// This returns false if we didn't expect the EndpointSlice to be
+		// deleted. If that is the case, we queue the Service for another sync.
+		if !c.endpointSliceTracker.Delete(endpointSlice) {
+			c.queueServiceForEndpointSlice(endpointSlice)
```

Comment:
what do you think about a metric here? is something you consider important or nice to have?
`endpointslice_stale_requeue_count`